pynamodb/attributes.py

```diff
@@ -272,6 +272,56 @@ def _set_attributes(self, **attributes):
                 raise ValueError("Attribute {0} specified does not exist".format(attr_name))
             setattr(self, attr_name, attr_value)
 
+    @staticmethod
+    def _serialize_value(attr, value, null_check=True):
+        """
+        Serializes a value for use with DynamoDB
+        :param attr: an instance of `Attribute` for serialization
+        :param value: a value to be serialized
+        :param null_check: If True, then attributes are checked for null
+        """
+        if value is None:
+            serialized = None
+        else:
+            serialized = attr.serialize(value)
+
+        if serialized is None:
+            if not attr.null and null_check:
+                raise ValueError("Attribute '{0}' cannot be None".format(attr.attr_name))
+            return {NULL: True}
+
+        return {ATTR_TYPE_MAP[attr.attr_type]: serialized}
+
+    @classmethod
+    def _serialize_container(cls, attr_container, values, null_check=True):
+        """
+        Retrieve attribute container recursively and do null check at the same time.
+        Handling for the raw MapAttribute will be done in MapAttribute
+        :param attr_container: subclass of AttributeContainer for serialization
+        :param values: value of the container to be serialized such as value of a model or MapAttribute, 
+        :param null_check: If True, then attributes are checked for null
+        """
+        attrs = {}
+        for name, attr in attr_container.get_attributes().items():
+            if not isinstance(values, dict) or not issubclass(attr_container, MapAttribute):
+                value = getattr(values, name)
+            else:
+                value = values.get(name)
+            if isinstance(value, MapAttribute) and type(value) is not MapAttribute:
+                serialized = {MAP_SHORT: cls._serialize_container(type(attr), value, null_check)}
+            else:
+                _null_check = null_check
+                if value is None and attr.default is not None:
+                    if attr.default:
+                        value = attr.default() if callable(attr.default) else attr.default
```

Comment:
What's the rationale behind assigning defaults here? Why do we need a second place to do this (the first being `_set_defaults`)?